esp/templates/users/userchecklist.html

```diff
@@ -17,10 +17,12 @@
 <script type="text/javascript">
     function checkAll(checkId){
         var inputs = document.querySelectorAll(checkId);
-        for (var i = 0; i < inputs.length; i++) {  
-            if (chk_new.checked) {
+        if (chk_new.checked) {
+            for (var i = 0; i < inputs.length; i++) {  
                 inputs[i].checked = true;
-            } else {
+            }
+        } else {
+            for (i = 0; i < inputs.length; i++) { 
                 inputs[i].checked = false;
             } 
```

Comment:
If you really want to be efficient, you could just have a single for loop where you set all inputs to the state of chk_new.checked, right?